lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -303,7 +304,12 @@ fCalibMJ := SEL(fFrequency <> 0, -9999, fPulseWattage / fFrequency);
         uHardWavelengthEdge3 .. 65535: bOverAllowableEnergy := fCalibMJ > 10;
     END_CASE
 {END_IF}
-FFO(i_xOK := NOT bOverAllowableEnergy,
+IF eEnumGet <> E_PPM_States.OUT AND bOverAllowableEnergy = TRUE THEN
+    uOverCounter := MIN(uOverCounter + 2, 50);
+ELSE
+    uOverCounter := MAX(uOverCounter - 1, 1);
+END_IF
```

Comment:
I like the counter approach here because it's easy to understand and doesn't do anything overly complex. This is what we want for a robust PLC program.

50 cycles at 10ms per cycle = 500ms = half a second
I'm not exactly sure what the "most optimal" count would be but this seems reasonable and will definitely cull the false positives.

Reply:
If bOverAllowableEnergy switches between True and False frequently, it could take up to 50 cycles or half a second to trigger the fault. (+2 * 50) + (-1 * 50) = 50, (+2 * 49) + (-1 * 51) = 47. As long as half of the cycles or more are over, it will trigger, but the higher the proportion the less time it takes, maxing at a quarter second. Like you said, these are all magic numbers that I just think seemed reasonable